### PR TITLE
ID-GNN example code

### PR DESCRIPTION
**Pull request by weihua916**
0.028 test MAP@12 on H&M as expected.

### CODE REVIEW

File: examples/model.py [hunk 1]
```diff
@@ -96,3 +105,35 @@ def forward(
...
+        dst_table: NodeType,
+    ) -> Tensor:
+        if self.id_awareness_emb is None:
+            raise RuntimeError(
+                "id_awareness must be set True to use forward_id_awareness"
+            )
+        seed_time = batch[entity_table].seed_time
+        x_dict = self.encoder(batch.tf_dict)
+        # Add ID-awareness to the root node
+        x_dict[entity_table][: seed_time.size(0)] += self.id_awareness_emb.weight
```
Comment (rusty1s) [inline examples/model.py:122]: This doesn't do anything for 2-hop GNNs? Do we need it?

File: relbench/external/loader.py [hunk 2]
```diff
@@ -41,11 +41,13 @@ def batched_arange(count: Tensor) -> Tuple[Tensor, Tensor]:
 class SparseTensor:
     r"""Sparse CSR tensor object that allows fast row tensor indexing."""
 
-    def __init__(self, sparse_tensor: Tensor):
+    def __init__(
+        self, sparse_tensor: Tensor, device: torch.device = torch.device("cpu")
```
Comment (rusty1s) [inline relbench/external/loader.py:45]: `None` is the better default, but not super critical.

File: examples/model.py [hunk 3]
```diff
@@ -96,3 +105,35 @@ def forward(
         )
 
         return self.head(x_dict[entity_table][: seed_time.size(0)])
+
+    def forward_id_awareness(
```
Comment (rusty1s) [inline examples/model.py:109]: I think `forward_id_awareness` is confusing. I assume you mean `forward_rhs_readout` or something like that.

File: examples/model.py (hunk 1 above)
Comment (weihua916) [reply examples/model.py:122]: No it does not do anything, but I still think it's good to add.

File: examples/model.py (hunk 3 above)
Comment (weihua916) [reply examples/model.py:109]: Sounds good.

**Pull request merged**